Add key cbindgen

prem-chand opened the pull request:
Please add the following dependency to the rosdep database.

## Package name:

`cbindgen`

## Package Upstream Source:

[https://github.com/mozilla/cbindgen](https://github.com/mozilla/cbindgen)

## Purpose of using this:

`cbindgen` is a tool that generates C header files from Rust source code. This is essential for enabling interoperability between Rust libraries/nodes and C/C++ components within the ROS ecosystem. Adding `cbindgen` to the rosdep database will allow ROS packages that utilize Rust to automatically install this build dependency, streamlining the development and build process for mixed-language ROS projects.

## Links to Distribution Packages

- Debian:
  - REQUIRED: [https://packages.debian.org/search?keywords=cbindgen](https://packages.debian.org/search?keywords=cbindgen) (e.g., `https://packages.debian.org/bullseye/cbindgen`)
- Ubuntu:
  - REQUIRED: [https://packages.ubuntu.com/search?keywords=cbindgen](https://packages.ubuntu.com/search?keywords=cbindgen) (e.g., `https://packages.ubuntu.com/focal/cbindgen`)
- Fedora:
  - IF AVAILABLE: [https://packages.fedoraproject.org/pkgs/rust-cbindgen/cbindgen/](https://packages.fedoraproject.org/pkgs/rust-cbindgen/cbindgen/)
- Arch:
  - IF AVAILABLE: [https://archlinux.org/packages/?name=cbindgen](https://archlinux.org/packages/?name=cbindgen)
- Gentoo:
  - IF AVAILABLE: [https://packages.gentoo.org/packages/dev-util/cbindgen](https://packages.gentoo.org/packages/dev-util/cbindgen)
- macOS (Homebrew):
  - IF AVAILABLE: [https://formulae.brew.sh/formula/cbindgen](https://formulae.brew.sh/formula/cbindgen)
- Alpine:
  - IF AVAILABLE: [https://pkgs.alpinelinux.org/packages?name=cbindgen](https://pkgs.alpinelinux.org/packages?name=cbindgen)
- NixOS/nixpkgs:
  - OPTIONAL: [https://search.nixos.org/packages?query=cbindgen](https://search.nixos.org/packages?query=cbindgen)
- openSUSE:
  - IF AVAILABLE: [https://software.opensuse.org/package/rust-cbindgen](https://software.opensuse.org/package/rust-cbindgen)
- RHEL (e.g., via EPEL or COPR):
  - IF AVAILABLE: `not available` (or specify if a known COPR or EPEL package exists, e.g., `rust-cbindgen` if present in EPEL for a specific RHEL version)

Commit:
added key cbindgen for ubuntu/debian

Commit:
Update cbindgen entries in base.yaml to use array format for package names

Commit:
Refactor cbindgen entries in base.yaml to use a simplified array format for Ubuntu and Debian

Commit:
Update cbindgen entries in base.yaml to specify Ubuntu focal and jammy versions

prem-chand committed:
Update base.yaml

Commit:
Implement feature X to enhance user experience and fix bug Y in module Z

Commit:
Implement feature X to enhance user experience and optimize performance

christophebedard commented on rosdep/base2.yaml:
I don't think adding this file is necessary.

christophebedard commented on rosdep/base.yaml:
```diff
@@ -495,6 +495,11 @@ catch2:
     '*': [catch2]
     focal: null
+cbindgen:
+  ubuntu:
+    focal: [cbindgen]
+    jammy: [cbindgen]
```
It's not actually in Ubuntu Focal, but it's in Jammy and later, so this should be:

```suggestion
  ubuntu:
    '*': [cbindgen]
    focal: null
```

https://packages.ubuntu.com/noble/cbindgen

christophebedard commented on rosdep/base.yaml:
```diff
@@ -495,6 +495,11 @@ catch2:
+  ubuntu:
+    focal: [cbindgen]
+    jammy: [cbindgen]
+  debian: [cbindgen]
```
The entry should be ordered alphabetically, so `debian` should be moved above `ubuntu`.

prem-chand replied:
entries reordered in alphabetical order

christophebedard reviewed (changes requested):
CI is also suggesting that this package is available for other OSes/distros:

```
Warning: Key 'cbindgen' might be satisfied by alpine package named 'cbindgen': https://pkgs.alpinelinux.org/package/edge/community/x86_64/cbindgen
Warning: Key 'cbindgen' might be satisfied by rhel package named 'cbindgen': https://ftp-nyc.osuosl.org/pub/fedora-epel/9/Everything/x86_64/Packages/c/cbindgen-0.27.0-1.el9.x86_64.rpm
Warning: Key 'cbindgen' might be satisfied by arch package named 'cbindgen': https://archlinux.org/packages/extra/x86_64/cbindgen/
Warning: Key 'cbindgen' might be satisfied by fedora package named 'cbindgen': https://mirror.umd.edu/fedora/linux/releases/42/Everything/x86_64/os/Packages/c/cbindgen-0.28.0-1.fc42.x86_64.rpm
```

https://github.com/ros/rosdistro/actions/runs/15037146879/job/42292496118?pr=45765

Can you take a look and add them if they're valid?

prem-chand committed:
Update rosdep/base.yaml

Co-authored-by: Christophe Bedard <bedard.christophe@gmail.com>

prem-chand committed:
Delete rosdep/base2.yaml

prem-chand committed:
Update base.yaml

christophebedard commented:
My comment above is still valid https://github.com/ros/rosdistro/pull/45765#pullrequestreview-2846918332

prem-chand committed:
Update base.yaml

CI suggested package is available for more OS/distros. Adding arch, alpine, fedora and rhel.

christophebedard commented on rosdep/base.yaml:
```diff
@@ -495,6 +495,15 @@ catch2:
+  arch: [cbindgen]
+  debian: [cbindgen]
+  fedora: [cbindgen]
+  rhel: [cbindgen]
```
CI says that this package is not available for RHEL 8, so this should probably be:

```suggestion
  rhel:
    '*': [cbindgen]
    '8': null
```

prem-chand committed:
Update rosdep/base.yaml

Co-authored-by: Christophe Bedard <bedard.christophe@gmail.com>

christophebedard reviewed (approved):
Thanks, looks good to me now. Note that we require two approvals for new rosdep keys.
